### PR TITLE
add origin weights

### PR DESCRIPTION
**Pull request by sthwang-metal**
# Summary of Changes

Add ability to set weight for an origin

### CODE REVIEW

**Review by tylerauerbeck — DISMISSED** [review]

File: db/migrations/20231017005257_add_origin_weight.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-- +goose Up
+-- modify "origins" table
+ALTER TABLE "origins" ADD COLUMN "weight" bigint NOT NULL DEFAULT 100;
```
Comment (rizzza) [inline db/migrations/20231017005257_add_origin_weight.sql:3]: `bigint` too large for the weights? Maybe `smallint`?

File: internal/ent/schema/origin.go [hunk 2]
```diff
@@ -40,6 +44,11 @@ func (Origin) Fields() []ent.Field {
 			Annotations(
 				entgql.OrderField("name"),
 			),
+		field.Int("weight").
```
Comment (rizzza) [inline internal/ent/schema/origin.go:47]: do we want to put any range constraints on the weight values that can persist?

Comment (tylerauerbeck) [reply internal/ent/schema/origin.go:47]: Yeah, I think you about covered it.Different providers accept it in all kinds of shapes and sizes, so putting too many constraints on this out of the gate will likely cause problems.

Comment (sthwang-metal) [reply internal/ent/schema/origin.go:47]: I can't seem to find any providers that use floats for weights, but in the worst case it should be pretty easy to convert from int to float database and code-wise

File: db/migrations/20231017005257_add_origin_weight.sql (hunk 1 above)
Comment (sthwang-metal) [reply db/migrations/20231017005257_add_origin_weight.sql]: seems like using smallint/int16 requires some manual implementations so I used int/int32 instead

**Review by rizzza — APPROVED** [review]

**Pull request merged**